registry/storage/driver/swift/swift.go

```diff
@@ -0,0 +1,519 @@
+// Swift.
+//
+// Because Swift is a key, value store the Stat call does not support last modification
+// time for directories (directories are an abstraction for key, value stores)
+package swift
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	gopath "path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ncw/swift"
+
+	"github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/base"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+)
+
+const driverName = "swift"
+
+const defaultChunkSize = 5 * 1024 * 1024
+
+//DriverParameters A struct that encapsulates all of the driver parameters after all values have been set
+type DriverParameters struct {
+	Username  string
+	Password  string
+	AuthURL   string
+	Tenant    string
+	Region    string
+	Container string
+	Prefix    string
+	ChunkSize int64
+}
+
+type swiftInfo map[string]interface{}
+
+func init() {
+	factory.Register(driverName, &swiftDriverFactory{})
+}
+
+// swiftDriverFactory implements the factory.StorageDriverFactory interface
+type swiftDriverFactory struct{}
+
+func (factory *swiftDriverFactory) Create(parameters map[string]interface{}) (storagedriver.StorageDriver, error) {
+	return FromParameters(parameters)
+}
+
+type driver struct {
+	Conn              swift.Connection
+	Container         string
+	Prefix            string
+	BulkDeleteSupport bool
+	ChunkSize         int64
+}
+
+type baseEmbed struct {
+	base.Base
+}
+
+// Driver is a storagedriver.StorageDriver implementation backed by Amazon Swift
+// Objects are stored at absolute keys in the provided bucket.
+type Driver struct {
+	baseEmbed
+}
+
+// FromParameters constructs a new Driver with a given parameters map
+// Required parameters:
+// - username
+// - password
+// - authurl
+// - container
+func FromParameters(parameters map[string]interface{}) (*Driver, error) {
+	username, ok := parameters["username"]
+	if !ok || fmt.Sprint(username) == "" {
+		return nil, fmt.Errorf("No username parameter provided")
+	}
+	password, ok := parameters["password"]
+	if !ok || fmt.Sprint(password) == "" {
+		return nil, fmt.Errorf("No password parameter provided")
+	}
+	authURL, ok := parameters["authurl"]
+	if !ok || fmt.Sprint(authURL) == "" {
+		return nil, fmt.Errorf("No container parameter provided")
+	}
+	container, ok := parameters["container"]
+	if !ok || fmt.Sprint(container) == "" {
+		return nil, fmt.Errorf("No container parameter provided")
+	}
+	tenant, ok := parameters["tenant"]
+	if !ok {
+		tenant = ""
+	}
+	region, ok := parameters["region"]
+	if !ok {
+		region = ""
+	}
+	rootDirectory, ok := parameters["rootdirectory"]
+	if !ok {
+		rootDirectory = ""
+	}
+	chunkSize := int64(defaultChunkSize)
+	chunkSizeParam, ok := parameters["chunksize"]
+	if ok {
+		chunkSize, ok = chunkSizeParam.(int64)
```

Comment:
@noxiouz I am familiar with this library. Considering the problems we've had with some of this `map[string]interface{}` code, it might be better to take on a dependency that handles this mapping in a consistent manner.